frontend/dockerfile/parser/directives.go

```diff
@@ -110,6 +112,47 @@ func (d *DirectiveParser) ParseAll(data []byte) ([]*Directive, error) {
 // This allows for a flexible range of input formats, and appropriate syntax
 // selection.
 func DetectSyntax(dt []byte) (string, string, []Range, bool) {
+	return detectDirective(keySyntax, dt)
+}
+
+func ParseCheckOptions(checkStr string) ([]string, bool, bool, error) {
+	parts := strings.SplitN(checkStr, ";", 2)
+	var skipSet []string
+	var errorOnWarn, skipAll bool
+	for _, p := range parts {
+		k, v, ok := strings.Cut(p, "=")
+		if !ok {
+			errors.Errorf("invalid check option %s", p)
+		}
+		k = strings.TrimSpace(k)
+		switch k {
+		case "skip":
+			v = strings.TrimSpace(v)
+			if v == "all" {
+				skipAll = true
+			} else {
+				skipSet = strings.Split(v, ",")
+				for i, rule := range skipSet {
+					skipSet[i] = strings.TrimSpace(rule)
+				}
+			}
+		case "error":
+			v = strings.TrimSpace(v)
+			errorOnWarn = v == "true"
+		default:
+			return nil, false, false, errors.Errorf("invalid check option %s", k)
+		}
+	}
+	return skipSet, skipAll, errorOnWarn, nil
+}
+
+// DetectCheck returns the check options of provided Dockerfile.
+// Follows all the same rules as the syntax directive.
+func DetectCheck(dt []byte) (string, string, []Range, bool) {
```

Comment:
Leave only `ParseDirectives()` to the parser package, and then parsing the value for "check" can be in dockerfile2llb helper.